lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -191,7 +191,83 @@ LocaleMatch lookupMatcher(
     const std::u16string &str) {
   return std::u16string(u"uppered");
 }
-
+// Implementation of
+// https://tc39.es/ecma402/#sec-getoption
+vm::CallResult<Option> getOption(
+    const Options &options,
+    const std::u16string &property,
+    const std::u16string &type,
+    const std::vector<std::u16string> &values,
+    const Options &fallback,
+    vm::Runtime *runtime) {
+  // 1. Assert type(options) is object
+  // 2. Let value be ? Get(options, property).
+  auto value = options.find(property);
+  // 3. If value is undefined, return fallback.
+  if (value == options.end()) {
+    auto fallbackIter = fallback.find(u"fallback");
+    // Make sure fallback exists
+    if (fallbackIter == options.end()) {
+      return runtime->raiseTypeError("Fallback does not exist");
+    }
+    return fallback.find(u"fallback")->second;
+  }
+  // 4. Assert: type is "boolean" or "string".
+  // 5. If type is "boolean", then
+  // a. Set value to ! ToBoolean(value).
+  // 6. If type is "string", then
+  // a. Set value to ? ToString(value).
+  if (type != u"string" && type != u"boolean") {
+    return runtime->raiseTypeError(vm::TwineChar16("Type ") + vm::TwineChar16(type.c_str()) + vm::TwineChar16(" is incorrect"));
+  }
+  // 7. If values is not undefined and values does not contain an element equal
+  // to value, throw a RangeError exception.
+  if (value->second.isString()) {
+    if (!values.empty() &&
+        llvh::find(values, value->second.getString()) == values.end()) {
+      return runtime->raiseRangeError(vm::TwineChar16("Value ") + vm::TwineChar16(value->second.getString().c_str()) + vm::TwineChar16(" out of range for Intl.DateTimeFormat options property " + vm::TwineChar16(property.c_str())));
+    }
+  } else {
+    if (!values.empty() &&
+        (value->second.getBool() != true && value->second.getBool() != false)) {
+      return runtime->raiseRangeError(vm::TwineChar16("Value ") + vm::TwineChar16(value->second.getString().c_str()) + vm::TwineChar16("  out of range for Intl.DateTimeFormat options property " + vm::TwineChar16(property.c_str())));
+    }
```

Comment:
I’m having trouble trying to get an error throw here if there is something that isn’t entered in as a bool i.e. hour12: “something”. Options is expecting a bool, so I think it just turns the value into True when using getBool/isBool because its kind is bool?